src/pages/confirmation/[id].tsx

```diff
@@ -75,19 +76,48 @@ export const getServerSideProps: GetServerSideProps = async (
   context: GetServerSidePropsContext
 ) => {
   const deviceType = deviceDetect(context.req.headers["user-agent"] ?? "");
+  
+  // Extract id from URL params
+  const { id:bookingID  } = context.params as { id: string };
```

Comment:
Because this is an API route id may not exist in the params so destructuring like this may cause a runtime error.
Better use optional chaining:

const bookingID = context.params?.id;